pubsub/gcloud/gcloud.go

```diff
@@ -70,10 +70,10 @@ func New(ctx context.Context, topic string, client *pubsub.Client, opts ...Optio
 // on a Google Cloud Pubsub topic.
 func (p *Gcloud) Publish(ctx context.Context, data []byte) error {
 	p.log.Debug().Msg("publishing message")
-	p.topic.Publish(ctx, &pubsub.Message{
+	_, err := p.topic.Publish(ctx, &pubsub.Message{
 		Data: data,
-	})
-	return nil
+	}).Get(ctx)
```

Comment:
According to the code docs. `Get` is blocking whilst `Publish` is not. 
```
// Get returns the server-generated message ID and/or error result of a Publish call.
// Get blocks until the Publish call completes or the context is done.
```
I'd be wary of making this change as it's a breaking change, applications may not be expecting the new blocking nature of the call. You could add a new method e.g. `PublishUntilComplete`

Reply:
On the other hand, all of our code that calls this function will be silently dropping errors.

I do agree with your point though that it could have unintended issues - I think we should add a task (maybe for the monthly maintenance next month?) to see if code can be swapped over to `PublishUntilComplete`.